### ISSUE TITLE
Screen setup

### ISSUE THREAD

**Issue by connyduck** [tuskyapp/Mod-companion#3]
(I never used the mastodon moderation interface, I definitely need some help with that)

Here is an overview of what the moderation api can do:
https://github.com/tootsuite/mastodon/pull/9387

Looks like the two main entities to work with are users and reports. So we probably need a way do display a list of users (search and sortable) and a list of reports (at least sortable), and a detail view for each where you can actually perform the moderation actions.

My first idea would be to just have a bottom bar that switches between the two lists and controls for sorting/searching in the top bar.

When I have more time I will draw some wireframes for this

**Comment by Briar503** [comment]
i was also thinking it would be neat to have a screen with instance analytics like (a line graph of toot activity, number of users, etc)

Also, would it make sense to add space in navigation for future mod api endpoints like silencing/suspending instances. If that number will likely go above three items, then maybe a nav drawer would make more sense? I suppose it could always be changed later though.

**Comment by connyduck** [comment]
> i was also thinking it would be neat to have a screen with instance analytics like (a line graph of toot activity, number of users, etc)

yes that would be awesome, there is no api support for it though. We should make issues at tootsuite for apis we would like to have.

> Also, would it make sense to add space in navigation for future mod api endpoints like silencing/suspending instances. If that number will likely go above three items, then maybe a nav drawer would make more sense? I suppose it could always be changed later though.

I'd say bottom nav is fine for up to 5 items, provided we can find good icons for them. Nav Drawer is definitive more flexible, for me both are ok.

**Comment by Briar503** [comment]
> yes that would be awesome, there is no api support for it though. We should make issues at tootsuite for apis we would like to have.

I was thinking we could just use the `instance` endpoint and grab the `stats` hash. If that is grabbed from a regular background task, then we can store change over time and use that for a line graph.
